test/services/oauth2/index.test.js

```diff
@@ -72,108 +75,77 @@ describe('oauth2 service', function oauthTest() {
 		key: '1',
 	};
 
-	let loginRequest1 = null;
-	let loginRequest2 = null;
 	// let redirectTo = null;
-
-	before((done) => {
+	const hydraUri = app.settings.services.hydra;
+	before(async () => {
 		this.timeout(10000);
-		Promise.all([
-			toolService.create(testTool1),
-			toolService.create(testTool2),
-		]).then(() => {
-			clientsService.create(testClient2).then((client) => {
-				const oauth = oauth2.create({
-					client: {
-						id: client.client_id,
-						secret: client.client_secret,
-					},
-					auth: {
-						tokenHost: 'http://localhost:9000',
-						tokenPath: '/oauth2/token',
-						authorizePath: '/oauth2/auth',
-					},
-				});
-				const authorizationUri = oauth.authorizationCode.authorizeURL({
-					redirect_uri: client.redirect_uris[0],
-					scope: 'openid',
-					state: '12345678',
-				});
-				request({
-					uri: authorizationUri,
-					method: 'GET',
-					followRedirect: false,
-				}).catch((res) => {
-					const position = res.error.indexOf('login_challenge=')
-							+ 'login_challenge'.length + 1;
-					loginRequest1 = res.error.substr(position, 32);
-					request({
-						uri: authorizationUri,
-						method: 'GET',
-						followRedirect: false,
-					}).catch((res2) => {
-						const position2 = res2.error.indexOf('login_challenge=')
-								+ 'login_challenge'.length + 1;
-						loginRequest2 = res2.error.substr(position2, 32);
-						done();
-					});
-				});
-			}).catch((err) => {
-				logger.warning('Can not execute oauth2 before all hook.', err);
-				done();
-			});
-		});
+
+		const o2mock = await oauth2Server({});
+		app.settings.services.hydra = o2mock.url;
+
+		app.configure(oauth2);
 	});
 
+
 	after((done) => {
-		Promise.all([
-			toolService.remove(testTool1._id),
-			toolService.remove(testTool2._id),
-			clientsService.remove(testClient2.client_id),
-		]).then(() => {
-			done();
-		}).catch((err) => {
-			logger.warning('Can not execute oauth2 after all hook.', err);
-			done();
-		});
+		// sets uri back to original uri
+		app.settings.services.hydra = hydraUri;
+		done();
 	});
 
 	it('is registered', () => {
 		assert.ok(clientsService);
 		assert.ok(loginService);
+		assert.ok(introspectService);
+		assert.ok(consentService);
 	});
 
 	it('GET BaseUrl', () => baseUrlService.find().then((response) => {
 		assert.ok(response);
 	}));
 
-	it('CREATE Client', () => clientsService.create(testClient).then((result) => {
+	it('CREATE Client', () => app.service('oauth2/clients').create(testClient).then((result) => {
```

Comment:
because the service is regenerated at runtime during before. 
It could however be considered to remove the clientsService constant at the top of the file, and consistently use app.service everywhere. (it is still being used in the first test)

Reply:
or alternatively, require the service in the before function after generating it. In that case, you have to disable the linting rule for that line... but its probably better overall